### DISCUSSION TITLE
How do plugin commucate with extension?

### DISCUSSION THREAD

**Discussion by SunnyFitz** [eclipse-theia/theia#9835] [Q&A]
I hava a backend plugin need some info from frontend extension。 I didn't find any way to do this .

**Comment by vince-fugnitto** [comment]
@SunnyFitz do you mind clarifying, is your plugin a vscode extension attempting to interact with a theia extension?

**Reply by SunnyFitz** [reply]
yes! i create a theia backend plugin and a theia extension , i want to exchange info between them ! how could i achieve it ?
Besides, is there anyway to exchange info between plugin and plugin 、extension and extension?

**Reply by vince-fugnitto** [reply]
@SunnyFitz I'm not aware if there is a mechanism for plugins and extensions to communicate, in the case of vscode plugins and theia extensions they can only communicate in a sense through commands.

@benoitf @RomanNikitenko do you have any additional information that can help?

**Reply by SunnyFitz** [reply]
Excute Command can pass arguments but  can't get any response . And those cause Initiator changed
